### DISCUSSION TITLE
How to set the number of wake planes in FAST.Farm to avoid early wake  truncation

### DISCUSSION THREAD

**Discussion by robytian** [OpenFAST/openfast#1171] [General]
Dear all，
During the simulation, I encountered the phenomenon as shown in the figure. Is this phenomenon wake truncation? Is there anything you can do to prevent this behavior? Or I want the wake to finish in a low resolution field.

 
![image](https://user-images.githubusercontent.com/86820784/177712193-a09db4ac-5330-4278-9836-ca59ab9d5483.png)
![image](https://user-images.githubusercontent.com/86820784/177712754-dbb2af18-a451-46ca-92f2-c3c26425efe6.png)

Best regards

**Comment by ebranlard** [comment]
Dear @robytian

What you observe is the fact that you reached the full number of wake planes (`NumPlanes`) that FAST.Farm stores in memory. If you increase this parameter, the wake will extent further downstream. Increasing the number of wake planes will increase the computational time, so that it's best to find a compromise such that the downstream turbines are reached (with a sufficient safety margin, of say half a wake length), with not too many wake planes. 

Typically, the number of wake planes needed will be a function of `DT_Low`, the mean wind speed `U0`, and the operating conditions of the turbine because a turbine under high thrust coefficient, will have a large induction in the wake and therefore the wake will propagate slower than the ambient wind speed. 

For instance, if the average axial induction factor of a given turbine is ,`a`, then the propagation speed of the wake is approximately `Uw=U0 (1-ka)` with `k=0.5` (diffusion will progressively make the wake propagate from `U0(1-2a)` to `U0`).  If the typical/maximal distance between turbines is `L = 10D`, adding a 50% margin, `L=15D`. The time to travel this distance is `T= L/Uw` and the number of wake planes is `NumPlanes = T/DT_Low = 15D/(U0 (1-k a)*DT_Low)`.  You can choose `a=0.33`. You will likely have to increase the safety margin on the distance based on your case. 

Modelling guidelines for FAST.Farm are found here:
https://openfast.readthedocs.io/en/dev/source/user/fast.farm/ModelGuidance.html

In particular, for wake dynamics: 
https://openfast.readthedocs.io/en/dev/source/user/fast.farm/ModelGuidance.html#wake-dynamics-parameters

I hope that helps,

Emmanuel
